src/region/regionUtil.ts

```diff
@@ -29,6 +45,97 @@ export const centerRegion = (shape: Shape): Position => {
     };
 };
 
+export const getPoints = (shape: Shape): [Point, Point, Point, Point] => {
+    const { height, width, x, y } = shape;
+
+    // Define the points from x,y and then in a clockwise fashion
+    // p1      p2
+    // +-------+
+    // |       |
+    // +-------+
+    // p4      p3
+    const p1 = { x, y };
+    const p2 = { x: x + width, y };
+    const p3 = { x: x + width, y: y + height };
+    const p4 = { x, y: y + height };
+
+    return [p1, p2, p3, p4];
+};
+
+export function selectTransformationPoint(shape: Shape, rotation: number): Point | null {
+    const [_p1, p2, p3, p4] = getPoints(shape);
+
+    // Determine which point will be the new x,y (as defined as the top left point) after rotation
+    // If -90deg: use p2
+    // If -180deg: use p3
+    // If -270deg: use p4
+    switch (rotation) {
+        case ROTATION_ONCE_DEG:
+            return p2;
+        case ROTATION_TWICE_DEG:
+            return p3;
+        case ROTATION_THRICE_DEG:
+            return p4;
+        default:
+            return null;
+    }
+}
+
+// Determines the translation needed to anchor the bottom left corner of the
+// coordinate space at the (0, 0) origin after rotation.
+export function selectTranslation(dimensions: Dimensions, rotation: number): Translation {
+    const { height, width } = dimensions;
+
+    switch (rotation) {
+        case ROTATION_ONCE_DEG:
+            return { dx: height };
+        case ROTATION_TWICE_DEG:
+            return { dx: width, dy: height };
+        case ROTATION_THRICE_DEG:
+            return { dy: width };
+        default:
+    }
+
+    return { dx: 0, dy: 0 };
+}
+
+export function getTransformedShape(
+    shape: Shape,
+    rotation: number,
+    dimensions: Dimensions = { height: 100, width: 100 },
```

Comment:
It looks like the `dimensions` argument is unused. Should we just make it a constant?